### PR TITLE
[3007.x] Add the ability to bootstrap a specific version of chocolatey

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Adds a state to bootstrap and unbootstrap chocolatey
Adds a changelog
Adds and updates tests

### What issues does this PR fix or reference?
Fixes: #64722

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by whytewolf — APPROVED** [review]

File: tests/pytests/functional/modules/test_chocolatey.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import pytest
+
+from salt.exceptions import CommandExecutionError
+
+pytestmark = [
+    pytest.mark.destructive,
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_chocolatey.py:6]:
```suggestion
    pytest.mark.destructive_test,
```
There is no `pytest.mark.destructive` mark

File: tests/pytests/functional/modules/test_chocolatey.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+@pytest.fixture()
+def clean(chocolatey):
+    chocolatey.unbootstrap()
+    try:
+        chocolatey_version = chocolatey.chocolatey_version(refresh=True)
+    except CommandExecutionError:
+        chocolatey_version = None
+    assert chocolatey_version is None
+    yield
+    chocolatey.unbootstrap()
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_chocolatey.py:27]:
```suggestion
def clean(chocolatey):
    chocolatey.unbootstrap()
    try:
        chocolatey_version = chocolatey.chocolatey_version(refresh=True)
    except CommandExecutionError:
        chocolatey_version = None
    assert chocolatey_version is None
    try:
        yield
    finally:
        chocolatey.unbootstrap()
```
We always want `chocolatey.unbootstrap()` to run, even if the test fails, correct?

File: tests/pytests/functional/states/chocolatey/test_bootstrap.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+@pytest.fixture()
+def clean(chocolatey_mod):
+    chocolatey_mod.unbootstrap()
+    try:
+        chocolatey_version = chocolatey_mod.chocolatey_version(refresh=True)
+    except CommandExecutionError:
+        chocolatey_version = None
+    assert chocolatey_version is None
+    yield
+    chocolatey_mod.unbootstrap()
```
Comment (s0undt3ch) [inline tests/pytests/functional/states/chocolatey/test_bootstrap.py:32]:
```suggestion
    try:
        yield
    finally:
        chocolatey_mod.unbootstrap()
```

File: tests/pytests/functional/states/chocolatey/test_bootstrap.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+@pytest.fixture()
+def installed(chocolatey_mod):
+    chocolatey_mod.bootstrap(force=True)
+    try:
+        chocolatey_version = chocolatey_mod.chocolatey_version(refresh=True)
+    except CommandExecutionError:
+        chocolatey_version = None
+    assert chocolatey_version is not None
+    yield
+    chocolatey_mod.unbootstrap()
```
Comment (s0undt3ch) [inline tests/pytests/functional/states/chocolatey/test_bootstrap.py:44]:
```suggestion
@pytest.fixture
def installed(chocolatey_mod):
    chocolatey_mod.bootstrap(force=True)
    try:
        chocolatey_version = chocolatey_mod.chocolatey_version(refresh=True)
    except CommandExecutionError:
        chocolatey_version = None
    assert chocolatey_version is not None
    try:
        yield
    finally:
        chocolatey_mod.unbootstrap()
```

**Review by s0undt3ch — DISMISSED** [review]

File: tests/pytests/functional/modules/test_chocolatey.py (hunk 1 above)
Comment (twangboy) [reply tests/pytests/functional/modules/test_chocolatey.py]: Done

File: tests/pytests/functional/modules/test_chocolatey.py (hunk 2 above)
Comment (twangboy) [reply tests/pytests/functional/modules/test_chocolatey.py]: Done

File: tests/pytests/functional/states/chocolatey/test_bootstrap.py (hunk 4 above)
Comment (twangboy) [reply tests/pytests/functional/states/chocolatey/test_bootstrap.py]: Done

File: tests/pytests/functional/states/chocolatey/test_bootstrap.py (hunk 3 above)
Comment (twangboy) [reply tests/pytests/functional/states/chocolatey/test_bootstrap.py]: Done

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**